src/lib/forms/validationAdapters/json-schema.js

```diff
@@ -0,0 +1,87 @@
+const jsonSchema = require('jsonschema');
+const { get } = require('lodash');
+
+const validate = (requestData, schema) => {
+  const result = jsonSchema.validate(requestData, schema);
+
+  return {
+    error: result.errors.length === 0 ? false : result,
+    value: requestData
+  };
+};
+
+const createSchemaFromForm = () => {
+  throw new Error('Forms are generated from schemas so not necessary to regenerate back from the form');
+};
+
+/**
+ * Determines the error key by inspecting the contents of
+ * the JSON schema validation error.
+ */
+const getErrorKey = error => {
+  const { property, argument } = error;
+  return property === 'instance' ? argument : property.replace('.instance', '');
+};
+
+/**
+ * For a given JSON schema validation error, a standard
+ * error object is returned containing the custom error
+ * text, if available, else the default JSON schema validation
+ * error text
+ */
+const getErrorMessages = (error, customErrors) => {
+  const key = getErrorKey(error);
+  const customError = customErrors[key];
+
+  if (customError && customError[error.name]) {
+    const { summary, message } = customError[error.name];
+    return { message, summary: summary || message };
+  }
+
+  return { message: error.message, summary: error.message };
+};
+
+const formatError = (error, customErrors) => {
+  const formattedError = {
+    name: getErrorKey(error),
+    ...getErrorMessages(error, customErrors)
+  };
+  return formattedError;
+};
+
+/**
+ * Copies the errors to the local form field level if
+ * there is an error from that field
+ */
+const applyLocalErrors = (form, errors) => {
+  form.fields.forEach(field => {
```

Comment:
Fields can contain child fields (in the case of conditional fields) so may need to use mapFields function to apply errors